notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+mnl_names_noncausal["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification_noncausal["household_size"] = [[2, 3]]
+mnl_names_noncausal["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification_noncausal["num_kids"] = [[2, 3]]
+mnl_names_noncausal["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_noncausal = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_noncausal,
+                                   model_type="MNL",
+                                   names=mnl_names_noncausal)
+
+num_vars_noncausal = len(reduce(lambda x, y: x + y, mnl_names_noncausal.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_noncausal.fit_mle(np.zeros(num_vars_noncausal),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_noncausal.get_statsmodels_summary()
+# -
+
+# ## Model 2
+
+# +
+mnl_specification_causal_1 = mnl_specification_noncausal.copy()
+mnl_names_causal_1 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_1["confounder_all"] = [1, 2, 3]
+mnl_names_causal_1["confounder_all"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_causal_1 = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_causal_1,
+                                   model_type="MNL",
+                                   names=mnl_names_causal_1)
+
+num_vars = len(reduce(lambda x, y: x + y, mnl_names_causal_1.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_causal_1.fit_mle(np.zeros(num_vars),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_causal_1.get_statsmodels_summary()
+# -
+
+# ## Model 3
+
+# +
+mnl_specification_causal_2 = mnl_specification_noncausal.copy()
+mnl_names_causal_2 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_2["confounder_all_2"] = [1, 2, 3]
+mnl_names_causal_2["confounder_all_2"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
```

Comment:
Same comments as above: I suggest refactoring the specification and name dictionary creation into its own file separate from this notebook.

Reply:
To show the difference between `mnl_specification_causal_1` and `mnl_specification_causal_2`, you can show `confounder_all_2` is in one specification but not the other (and vice versa for `confounder_all_1`).